Fortran/gfortran/README.md

```diff
@@ -129,8 +129,41 @@ implemented at a steady pace. The relevant tests in this directory should be
 enabled. This would involve building the test suite with one of the
 `TEST_SUITE_FORTRAN_*` flags described above.
 
-The test files should be kept in sync with gfortran. This needs to be done
-manually periodically.
+_*NOTE*: We are currently in the process of migrating to the static test 
+configuration described below. During that time, the configuration files and 
+associated scripts will be present in the repository but will not be used. When
+the migration is completed, this note will be removed._
+
+The build system uses static test configuration files named `tests.cmake` to be 
+found in the various subdirectories of the test suite. These are generated by 
+`utils/update-test-config.py`. The configuration files are the result of parsing
+the relevant DejaGNU annotations from the test files and are used by the various
+`CMakeLists.txt` files to set up the tests. This was a lot easier --- and 
+cleaner --- than parsing the annotations from `cmake`. 
+
+The test configuration files consist of comment and non-comment lines. Comment
+lines start with a `#`. Each non-comment line represents a single test. The line
+consists of semicolon-separated fields as shown below:
+
+```<kind>;<sources>;<xfail>;<options>;<enabled-on>;<disabled-on>```
+
+Each field is described in the table below:
+
+| Field          | Description
+| :--------------|:----------------------------------------------------------
+| `<kind>`       | Must be one of 'preprocess', 'assemble', 'compile', 'link' or 'execute'.
+| `<sources>`    | A space separated list of sources files that comprise the test. The first file is the "main" file. The rest of the files must be specified in program compilation order
```

Comment:
Are the paths relative or absolute?

Separating the paths using spaces feels risky because paths themselves might contain spaces, especially on Windows systems. I guess if the paths are relative then we can control whether they have spaces within this repository, but this definitely isn't safe for absolute paths.

Reply:
Sounds good to me